### ISSUE TITLE
Error getting write for arrays to work

### ISSUE THREAD

**Issue by bobdonat** [mikefarah/yq#622]
I am trying to create a yaml that looks like the following:

``` yaml
a:
  test: foo
  b:
  - value: "false"
```
 
This is the command I am executing.  It is not clear to me what is wrong.
./yq -i eval '.a.b[0].value = "false"' /scratch/test.yaml
Error: expected end of expression but found '=', please check expression syntax

Version of yq: 4.1.0
Operating system: mac
Installed via: go get github.com/mikefarah/yq/v4

**Input Yaml**
```yaml
a:
  test: foo
```

**Comment by mikefarah** [comment]
Looks like a problem with my token parsing for array indexes, I'll fix it in the next release, for now you can do:

```
./yq -i eval '.a.b.[0].value = "false"' /scratch/test.yaml
```

(note the dot between `b` and `[0]`)

**Comment by bobdonat** [comment]
Thanks, that workaround works fine.  Will this workout still be valid when 4.1.1 is released?  Do you have a time frame for 4.1.1.?

**Comment by mikefarah** [comment]
Yep it will still work in 4.1.1, hopefully will release within this week :)
________________________________

**Comment by bobdonat** [comment]
Different issue with env variables.

I have:
export FOO=some-value
./yq -i eval '.a.b.[0].value = "$FOO"' /scratch/test.yaml

I do not see the value for env variable FOO instead I get:
...
      - value: "$FOO"

**Comment by mikefarah** [comment]
That is a different issue - please raise separate issues as they make issue tracking much easier. Single quotes in bash do not interpolate strings, you'll need to use double quotes (and escape the double quotes inside) for it to work
".a.b.[0].value = \"$FOO\""

**Comment by mikefarah** [comment]
Fixed in 4.2.0 - note that both forms work `.a[0]` as well as `.a.[0]`
